tests/test_solving.py

```diff
@@ -0,0 +1,81 @@
+from anonlink.solving import greedy_solve
+
+def _zip_candidates(candidates):
+    candidates = tuple(candidates)
+    if not candidates:
+        return ((), ()), ((), ()), ()
+    records1, records2, sims = zip(*candidates)
+    dataset_indices = []
+    record_indices = []
+    for (d1, r1), (d2, r2) in zip(records1, records2):
+        dataset_indices.append((d1, d2))
+        record_indices.append((r1, r2))
+    return (tuple(zip(*dataset_indices)),
+            tuple(zip(*record_indices)),
+            sims)
+
+
+def _compare_matching(result, truth):
+    result = frozenset(map(frozenset, result.values()))
+    truth = frozenset(map(frozenset, truth))
+    assert result == truth
+
+
+def test_greedy_twoparty():
```

Comment:
This test and the one below could be combined into a single tests that is parameterised on pairs containing `candidates` and the corresponding expected result.

Reply:
Good point. I don’t want to spend any more time on this PR, but I’ll refactor these tests before I merge into master.